crates/notion-core/src/toolchain/mod.rs

```diff
@@ -0,0 +1,93 @@
+use std::fs::File;
+use std::io::Write;
+
+use readext::ReadExt;
+use semver::Version;
+use toml;
+
+use distro::node::NodeVersion;
+use fs::touch;
+use image::Image;
+use path::user_platform_file;
+
+use notion_fail::{Fallible, ResultExt};
+
+pub(crate) mod serial;
+
+pub struct Toolchain {
+    platform: Option<Image>,
+}
+
+impl Toolchain {
+    pub fn current() -> Fallible<Toolchain> {
+        let path = user_platform_file()?;
+        let src = touch(&path)?.read_into_string().unknown()?;
+        let serial: serial::Platform = toml::from_str(&src).unknown()?;
+        Ok(Toolchain {
+            platform: serial.into_image()?
+        })
+    }
+
+    pub fn get_installed_node(&self) -> Option<NodeVersion> {
```

Comment:
It seems like `installed` might not be the right adjective here, since technically all of the versions in the inventory are "installed" in the sense that they are available without a download.

Perhaps something like `active` e.g. `get_active_node`?

Reply:
I can see how someone reading the code could find that ambiguous. "Active" sounds pretty good to me!